spec/rubocop/cop/rspec/spec_file_path_format_spec.rb

```diff
@@ -299,12 +299,40 @@ class Foo
     end
   end
 
-  context 'when configured with `CustomTransform: { "FooFoo" => "foofoo" }`' do
-    let(:cop_config) { { 'CustomTransform' => { 'FooFoo' => 'foofoo' } } }
+  context 'when configured with `CustomTransform: { "RSpec" => "rspec" }`' do
+    let(:cop_config) { { 'CustomTransform' => { 'RSpec' => 'rspec' } } }
+    let(:suffix) { 'foo/some/r_spec_foo*bar*_spec.rb' }
 
-    it 'does not register an offense for custom module name transformation' do
-      expect_no_global_offenses(<<-RUBY, 'foofoo/some/class/bar_spec.rb')
-        describe FooFoo::Some::Class, '#bar' do; end
+    it 'registers an offense when not an exact match to custom ' \
+       'module name transformation' do
+      expect_global_offense(<<-RUBY, 'foo/some/rspec_foo/bar_spec.rb', message)
+        describe Foo::Some::RSpecFoo, '#bar' do; end
+      RUBY
+    end
+
+    it 'does not register an offense when an exact match to custom ' \
+       'module name transformation' do
+      expect_no_global_offenses(<<-RUBY, 'rspec/some/foo/bar_spec.rb')
+        describe RSpec::Some::Foo, '#bar' do; end
+      RUBY
+    end
+  end
+
+  context 'when configured with `CustomTransformPatterns: ' \
+          '{ "RSpec" => "rspec" }`' do
+    let(:cop_config) { { 'CustomTransformPatterns' => { 'RSpec' => 'rspec' } } }
```

Comment:
What i meant with reusing the option was to allow ‘“*RSpec*” => “rspec”’ config in CustomTransform.
That would still be backwards-compatible.
I’m not against the new option, but I experience some cognitive resistance due to the lack of real examples.
Would the following make good examples of partial trasgormations?
API -> api
DSL -> dsl

Reply:
I’m still not convinced with RSpecClass.